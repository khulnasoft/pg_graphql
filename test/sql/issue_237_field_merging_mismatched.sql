begin;
    -- https://github.com/khulnasoft/pg_graphql/issues/237
    savepoint a;
    create table blog_post(
        id int primary key,
        a text,
        b text,
        c text,
        d text,
        e text,
        f text
    );
    insert into public.blog_post
    values (1, 'a', 'b', 'c', 'd', 'e', 'f');
    -- mismatched field names
    select jsonb_pretty(
      graphql.resolve($$
        query {
          blogPostCollection {
            edges {
              node {
                a
              }
            }
          }
          blogPostCollection {
            edges {
              node {
                a: b
              }
            }
          }
        }
      $$)
    );
    -- mismatched arguments
    select jsonb_pretty(
      graphql.resolve($$
        query {
          blogPostCollection(filter: {
            id: { eq: 1 }
          }) {
            edges {
              node {
                a
              }
            }
          }
          blogPostCollection {
            edges {
              node {
                b
              }
            }
          }
        }
      $$)
    );
    -- mismatched list to node
    select jsonb_pretty(
      graphql.resolve($$
        query {
          blogPostCollection {
            a: edges {
              cursor
            }
          }
          blogPostCollection {
            a: pageInfo {
              cursor: endCursor
            }
          }
        }
      $$)
    );

    rollback to savepoint a;

    create table account(
        id serial primary key,
        email varchar(255) not null
    );

    insert into public.account(email)
    values
        ('aardvark@x.com');

    create table blog(
        id serial primary key,
        owner_id integer not null references account(id),
        name varchar(255) not null
    );

    insert into blog(owner_id, name)
    values
        (1, 'A: Blog 1');

    select jsonb_pretty(graphql.resolve($$ {
        accountCollection {
            edges {
                node {
                    email: id
                    email
                }
            }
        }
    }$$));

    select jsonb_pretty(graphql.resolve($$ {
        accountCollection(first: 1) {
            edges {
                node {
                    id
                    email
                }
            }
        }
        accountCollection(first: 2) {
            edges {
                node {
                    id
                    email
                }
            }
        }
    }$$));

    select jsonb_pretty(graphql.resolve($$ {
        accountCollection(first: $count) {
            edges {
                node {
                    id
                    email
                }
            }
        }
        accountCollection(first: 1) {
            edges {
                node {
                    id
                    email
                }
            }
        }
    }$$,
    jsonb_build_object(
        'count', 1
    )));

    select jsonb_pretty(graphql.resolve($$ {
        accountCollection(first: $count) {
            edges {
                node {
                    id
                    email
                }
            }
        }
        accountCollection(first: $num) {
            edges {
                node {
                    id
                    email
                }
            }
        }
    }$$,
    jsonb_build_object(
        'count', 1,
        'num', 1
    )));

    select jsonb_pretty(graphql.resolve($$ {
        accountCollection(first: 1) {
            edges {
                node {
                    id
                    email
                }
            }
        }
        accountCollection {
            edges {
                node {
                    id
                    email
                }
            }
        }
    }$$));

rollback;
