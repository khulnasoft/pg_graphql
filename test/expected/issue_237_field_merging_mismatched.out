begin;
    -- https://github.com/khulnasoft/pg_graphql/issues/237
    savepoint a;
    create table blog_post(
        id int primary key,
        a text,
        b text,
        c text,
        d text,
        e text,
        f text
    );
    insert into public.blog_post
    values (1, 'a', 'b', 'c', 'd', 'e', 'f');
    -- mismatched field names
    select jsonb_pretty(
      graphql.resolve($$
        query {
          blogPostCollection {
            edges {
              node {
                a
              }
            }
          }
          blogPostCollection {
            edges {
              node {
                a: b
              }
            }
          }
        }
      $$)
    );
                       jsonb_pretty                        
-----------------------------------------------------------
 {                                                        +
     "data": null,                                        +
     "errors": [                                          +
         {                                                +
             "message": "Fields `b` and `a` are different"+
         }                                                +
     ]                                                    +
 }
(1 row)

    -- mismatched arguments
    select jsonb_pretty(
      graphql.resolve($$
        query {
          blogPostCollection(filter: {
            id: { eq: 1 }
          }) {
            edges {
              node {
                a
              }
            }
          }
          blogPostCollection {
            edges {
              node {
                b
              }
            }
          }
        }
      $$)
    );
                                      jsonb_pretty                                       
-----------------------------------------------------------------------------------------
 {                                                                                      +
     "errors": [                                                                        +
         {                                                                              +
             "message": "Two fields named `blogPostCollection` have different arguments"+
         }                                                                              +
     ]                                                                                  +
 }
(1 row)

    -- mismatched list to node
    select jsonb_pretty(
      graphql.resolve($$
        query {
          blogPostCollection {
            a: edges {
              cursor
            }
          }
          blogPostCollection {
            a: pageInfo {
              cursor: endCursor
            }
          }
        }
      $$)
    );
                             jsonb_pretty                             
----------------------------------------------------------------------
 {                                                                   +
     "data": null,                                                   +
     "errors": [                                                     +
         {                                                           +
             "message": "Fields `pageInfo` and `edges` are different"+
         }                                                           +
     ]                                                               +
 }
(1 row)

    rollback to savepoint a;
    create table account(
        id serial primary key,
        email varchar(255) not null
    );
    insert into public.account(email)
    values
        ('aardvark@x.com');
    create table blog(
        id serial primary key,
        owner_id integer not null references account(id),
        name varchar(255) not null
    );
    insert into blog(owner_id, name)
    values
        (1, 'A: Blog 1');
    select jsonb_pretty(graphql.resolve($$ {
        accountCollection {
            edges {
                node {
                    email: id
                    email
                }
            }
        }
    }$$));
                          jsonb_pretty                          
----------------------------------------------------------------
 {                                                             +
     "data": null,                                             +
     "errors": [                                               +
         {                                                     +
             "message": "Fields `email` and `id` are different"+
         }                                                     +
     ]                                                         +
 }
(1 row)

    select jsonb_pretty(graphql.resolve($$ {
        accountCollection(first: 1) {
            edges {
                node {
                    id
                    email
                }
            }
        }
        accountCollection(first: 2) {
            edges {
                node {
                    id
                    email
                }
            }
        }
    }$$));
                                      jsonb_pretty                                      
----------------------------------------------------------------------------------------
 {                                                                                     +
     "errors": [                                                                       +
         {                                                                             +
             "message": "Two fields named `accountCollection` have different arguments"+
         }                                                                             +
     ]                                                                                 +
 }
(1 row)

    select jsonb_pretty(graphql.resolve($$ {
        accountCollection(first: $count) {
            edges {
                node {
                    id
                    email
                }
            }
        }
        accountCollection(first: 1) {
            edges {
                node {
                    id
                    email
                }
            }
        }
    }$$,
    jsonb_build_object(
        'count', 1
    )));
                                      jsonb_pretty                                      
----------------------------------------------------------------------------------------
 {                                                                                     +
     "errors": [                                                                       +
         {                                                                             +
             "message": "Two fields named `accountCollection` have different arguments"+
         }                                                                             +
     ]                                                                                 +
 }
(1 row)

    select jsonb_pretty(graphql.resolve($$ {
        accountCollection(first: $count) {
            edges {
                node {
                    id
                    email
                }
            }
        }
        accountCollection(first: $num) {
            edges {
                node {
                    id
                    email
                }
            }
        }
    }$$,
    jsonb_build_object(
        'count', 1,
        'num', 1
    )));
                                      jsonb_pretty                                      
----------------------------------------------------------------------------------------
 {                                                                                     +
     "errors": [                                                                       +
         {                                                                             +
             "message": "Two fields named `accountCollection` have different arguments"+
         }                                                                             +
     ]                                                                                 +
 }
(1 row)

    select jsonb_pretty(graphql.resolve($$ {
        accountCollection(first: 1) {
            edges {
                node {
                    id
                    email
                }
            }
        }
        accountCollection {
            edges {
                node {
                    id
                    email
                }
            }
        }
    }$$));
                                      jsonb_pretty                                      
----------------------------------------------------------------------------------------
 {                                                                                     +
     "errors": [                                                                       +
         {                                                                             +
             "message": "Two fields named `accountCollection` have different arguments"+
         }                                                                             +
     ]                                                                                 +
 }
(1 row)

rollback;
